docs/cli/dotnet.rst

```diff
@@ -1,8 +1,161 @@
+
+The resulting project.json should include these items (in addition to your other project dependencies).
+
+.. code-block:: json
+
+    {
+        "dependencies": {
+            "Microsoft.EntityFrameworkCore.Tools": {
+                "type": "build",
+                "version": "1.0.0-preview1"
+            }
+        },
+
+        "tools": {
+            "Microsoft.EntityFrameworkCore.Tools": {
+                "imports": ["portable-net451+win8"],
+                "version": "1.0.0-preview1"
+            }
+        },
+
+        "frameworks": {
+            "netcoreapp1.0": {
+                "imports": "portable-net451+win8"
+            }
+        }
+    }
+
+.. tip::
+    A build-only dependency (``"type": "build"``) means this dependency is local to the current project. For example, if Project A has a build only dependency and Project B depends on A, ``dotnet restore`` will not add A's build-only dependencies into Project B.
+
+Usage
+-----
+Commands can be run from the command line by navigating to the project directory and executing ``dotnet ef [subcommand]``. To see usage, add ``--help`` to any command to see more information about parameters and subcommands.
+
+.. TODO add usage here
+
+Preview 1 Known Issues
+----------------------
+
+Targeting class library projects is not supported
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.NET Core CLI does not support running commands on class libraries as of Preview 1. Despite being able to install EF tools, executing commands may throw this error message.
+
+.. code-block:: text
+
+    This preview of Entity Framework tools does not support targeting class library projects in ASP.NET Core and .NET Core applications.
+
+See issue https://github.com/dotnet/cli/issues/2645. 
+
+Workaround
+^^^^^^^^^^
+Convert your class library project into an "app" project. This can either be a .NET Core app or a desktop .NET app.
+
+To make the project a .NET Core App, add the "netcoreapp1.0" framework to project.json along with the other settings in the sample below:
+
+.. code-block:: json
+
+    {
+        "frameworks": {
+            "netcoreapp1.0": {
+                "imports": ["portable-net451+win8"],
+                "buildOptions": {
+                    "emitEntryPoint": true
+                },
+                "dependencies": {
+                    "Microsoft.NETCore.App": {
+                        "type": "platform",
+                        "version": "1.0.0-*"
+                    }
+                }
+            }
+        }
+    }
+
+To make a desktop .NET app, ensure you project targets "net451" or newer (example "net461" also works) and ensure the build option ``"emitEntryPoint"`` is set to true.
+
+.. code-block:: json
+
+    {
+        "frameworks": {
+            "net451": {
+                "buildOptions": {
+                    "emitEntryPoint": true
+                }
+            }
+        }
+    }
+
```

Comment:
Added a small sample of using IDbContextFactory to this PR. See https://github.com/aspnet/EntityFramework.Docs/pull/159/commits/120d3a686beb3d977aa052367c74d348479b00b6